From c4eb39966cfdc4696b263703f58f0ab8cecc7973 Mon Sep 17 00:00:00 2001
From: Bin Yang <yangbin@rock-chips.com>
Date: Wed, 6 Nov 2019 16:34:10 +0800
Subject: [PATCH 25/25] mtp: Define _FILE_OFFSET_BITS=64 to support 2~4G file

File offset type "off_t" is a signed long integer, so the maximum
offset is 2GB. If the file size is bigger than 2GB, the file offset
will be type-cast to a negative number. And it will cause a file is
limited to 2GB or MTP transmission break.

Test: manual - connect Android device to Windowns PC
Test: manual - select MTP/File Transfer in android device
Test: manual - copy a file which is bigger than 2GB from Windowns PC to
               the android device.

Change-Id: I5c7935befa72abe28c832b8022cac4bc9e4ae4ac
Signed-off-by: Bin Yang <yangbin@rock-chips.com>
(cherry picked from commit 8e1978aec719ba7de94687b36f346e78b8879fa5)
---
 media/mtp/Android.bp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/media/mtp/Android.bp b/media/mtp/Android.bp
index 66a31392fd..f0f65d92a9 100644
--- a/media/mtp/Android.bp
+++ b/media/mtp/Android.bp
@@ -40,6 +40,7 @@ cc_library_shared {
     ],
     export_include_dirs: ["."],
     cflags: [
+	"-D_FILE_OFFSET_BITS=64",
         "-DMTP_DEVICE",
         "-DMTP_HOST",
         "-Wall",
-- 
2.35.1

